processing/src/main/java/org/apache/druid/query/scan/ScanQueryQueryToolChest.java

```diff
@@ -187,11 +203,112 @@ public RowSignature resultArraySignature(final ScanQuery query)
     }
   }
 
+  /**
+   * This batches the fetched {@link ScanResultValue}s which have similar signatures and are consecutives. In best case
+   * it would return a single frame, and in the worst case, it would return as many frames as the number of results
```

Comment:
```suggestion
   * it would return a single frame, and in the worst case, it would return as many frames as the number of {#link ScanResultValue}
```